src/argilla/client/feedback/dataset.py

```diff
@@ -568,66 +642,79 @@ def push_to_argilla(
+                    ).parsed
+                    if self.argilla_id is None:
+                        field.id = new_field.id
                 except Exception as e:
                     delete_dataset(dataset_id=argilla_id)
                     raise Exception(
-                        f"Failed while adding the field '{field.name}' to the `FeedbackTask` dataset in Argilla with"
+                        f"Failed while adding the field '{field.name}' to the `FeedbackDataset` in Argilla with"
                         f" exception: {e}"
                     ) from e
 
-            for question in self.questions:
+            question_name2id = {}
+            for question in self.__questions:
                 try:
-                    datasets_api_v1.add_question(client=httpx_client, id=argilla_id, question=question.dict())
+                    new_question = datasets_api_v1.add_question(
+                        client=httpx_client, id=argilla_id, question=question.dict(exclude={"id"})
+                    ).parsed
+                    if self.argilla_id is None:
+                        question.id = new_question.id
+                    question_name2id[new_question.name] = new_question.id
                 except Exception as e:
                     delete_dataset(dataset_id=argilla_id)
                     raise Exception(
-                        f"Failed while adding the question '{question.name}' to the `FeedbackTask` dataset in Argilla"
+                        f"Failed while adding the question '{question.name}' to the `FeedbackDataset` in Argilla"
                         f" with exception: {e}"
                     ) from e
 
             try:
                 datasets_api_v1.publish_dataset(client=httpx_client, id=argilla_id)
             except Exception as e:
                 delete_dataset(dataset_id=argilla_id)
-                raise Exception(
-                    f"Failed while publishing the `FeedbackTask` dataset in Argilla with exception: {e}"
-                ) from e
+                raise Exception(f"Failed while publishing the `FeedbackDataset` in Argilla with exception: {e}") from e
 
-            for batch in self.iter():
+            for i in trange(
+                0, len(self.records), PUSHING_BATCH_SIZE, desc="Pushing records to Argilla...", disable=show_progress
+            ):
                 try:
+                    records = []
+                    for record in self.records[i : i + PUSHING_BATCH_SIZE]:
+                        if record.suggestions:
+                            for suggestion in record.suggestions:
+                                suggestion.question_id = question_name2id[suggestion.question_name]
+                        records.append(
+                            record.dict(
+                                exclude={"id": ..., "suggestions": {"__all__": {"question_name"}}}, exclude_none=True
+                            )
+                        )
```

Comment:
this is duplicated code, maybe we can refactor a bit more to avoid some code duplication

Reply:
I'll refactor everything in an upcoming PR to create `ArgillaMixin` to add all those things there instead 👍🏻